Source/main.cpp

```diff
@@ -188,43 +227,70 @@ int main(void)
         lag_hyd_starboard = hyd_starboard.get_lag();
         lag_hyd_stern = hyd_stern.get_lag();
 
-        // Checking if the lag is valid
-        // Take new sample if not valid data
-        if(!TRILITERATION::check_valid_signals(lag_hyd_port,
-              lag_hyd_starboard, lag_hyd_stern, 0, 0, 0)){
-          log_error(Error_types::ERROR_INVALID_SIGNAL)
-          continue;
-        }
-
-        // Calculate an estimate for the range
         intensity_port = hyd_port.get_intensity();
         intensity_starboard = hyd_starboard.get_intensity();
         intensity_stern = hyd_stern.get_intensity();
 
-        range_es_port = hyd_port.get_lag();
-        range_es_starboard = hyd_starboard.get_lag();
-        range_es_stern = hyd_stern.get_lag();
+        /**
+         * Checking is the measurements are valid. The measurements 
+         * are discarded if they deviate too much in either signal
+         * intensity or time lag
+         * 
+         * Take new samples if the data is invalid
+         */
+        if(!TRILITERATION::check_valid_signals(lag_hyd_port,
```

Comment:
I notice that a few functions are taking all the lags and intensities for each hydrophone, which amounts to 6 parameters. It would potentially be cleaner to pass them as arrays,  i.e uint32_t lags[NUM_HYDROPHONES] = {portside, startboard, stern} and uint32_t intensities[NUM_HYDROPHONES] = {portside, startboard, stern}